src/tooltips.ts

```diff
@@ -0,0 +1,244 @@
+import isString = require('lodash/isString');
+import noop = require('lodash/noop');
+import { TooltipPlacement as Placement } from 'rendition';
+import { px } from './utils';
+
+// The distance between the tooltip arrow and the target element
+const TARGET_OFFSET = 3;
+
+// The time in ms to show a tooltip after being triggered by a click event
+const CLICK_TIMEOUT = 1000;
+
+const arrowStyle = `
+	position: absolute;
+	width: 0;
+	height: 0;
+	border-color: transparent;
+	border-style: solid;
+`;
+
+const topArrowStyle = `
+	${arrowStyle}
+	bottom: 0;
+	left: 50%;
+	margin-left: -5px;
+	border-width: 5px 5px 0;
+	border-top-color: #000;
+`;
+
+const rightArrowStyle = `
+	${arrowStyle}
+	top: 50%;
+	left: 0;
+	margin-top: -5px;
+	border-width: 5px 5px 5px 0;
+	border-right-color: #000;
+`;
+
+const leftArrowStyle = `
+	${arrowStyle}
+	top: 50%;
+	right: 0;
+	margin-top: -5px;
+	border-width: 5px 0 5px 5px;
+	border-left-color: #000;
+`;
+
+const bottomArrowStyle = `
+	${arrowStyle}
+	top: 0;
+  left: 50%;
+  margin-left: -5px;
+  border-width: 0 5px 5px;
+  border-bottom-color: #000;
+`;
+
+const getArrowStyle = (placement?: Placement) => {
+	switch (placement) {
+		case 'right':
+			return rightArrowStyle;
+		case 'bottom':
+			return bottomArrowStyle;
+		case 'left':
+			return leftArrowStyle;
+		default:
+			return topArrowStyle;
+	}
+};
+
+export class Tooltips {
+	private tooltipElement: HTMLDivElement;
+	private tooltipElementInner: HTMLDivElement;
+	private tooltipElementArrow: HTMLDivElement;
+	private hideTimeout: number;
+	private initialised: boolean = false;
+
+	initialiseElements() {
+		if (this.initialised || !document || !document.body) {
+			return;
+		}
+
+		const tooltipElement = document.createElement('div');
```

Comment:
This is interesting. Why can't we use React + styled components to do all this tooltip stuff?

You probably don't want to put it in the downstream tree, but you could define a component here and separately call ReactDOM.render() with it to put it in the page yourself. I think that should still let you do this The React Way, and make a lot of this much simpler.

Reply:
Good idea